src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,305 @@
+   */
+  public updateWidthOfSuggestionContainer(container: Dom) {
+    if (!this.options.suggestionWidth) {
+      return;
+    }
+    container.el.style.minWidth = this.options.suggestionWidth;
+    container.el.style.maxWidth = this.options.suggestionWidth;
+  }
+
+  private autoCreateComponentsInsideResult(element: HTMLElement, result: IQueryResult): IInitResult {
+    Assert.exists(element);
+    return Initialization.automaticallyCreateComponentsInsideResult(element, result);
+  }
+
+  private buildPreviewHeader(suggestion: string) {
+    const text = $$('span', {}, `${this.options.headerText} "${suggestion}"`).el;
+    const header = $$('div', { className: 'coveo-preview-header' }, text).el;
+    this.previewContainer.appendChild(header);
+  }
+
+  private buildResultsContainer() {
+    const resultsContainer = $$('div', { className: 'coveo-preview-results' }).el;
+    if (!this.options.previewWidth) {
+      return resultsContainer;
+    }
+
+    resultsContainer.style.width = `${this.options.previewWidth}px`;
+    return resultsContainer;
+  }
+
+  private get previewContainer() {
+    return $$(this.root).find('.coveo-preview-container');
   }
 
   private querySuggestGetFocus(args: IQuerySuggestSelection) {
-    if (!this.shouldShowPreviewResults) {
+    if (!this.previewContainer) {
       return;
     }
-    this.executeQueryHover(args.suggestion);
+
+    if (this.previousSuggestionHovered === args.suggestion) {
+      return;
+    }
+
+    this.previousSuggestionHovered = args.suggestion;
+    this.timer && clearTimeout(this.timer);
+    this.timer = setTimeout(() => {
+      this.executeQueryHover(args.suggestion);
+    }, this.options.hoverTime);
   }
 
   private executeQueryHover(suggestion: string) {
     const previousQueryOptions = this.queryController.getLastQuery();
     previousQueryOptions.q = suggestion;
     previousQueryOptions.numberOfResults = this.options.numberOfPreviewResults;
-    //TODO: I will need to execute a query, with the result,
-    //      build a container and display the result  next to the querySuggest
+    this.queryController
+      .getEndpoint()
+      .search(previousQueryOptions)
+      .then(results => {
+        $$(this.previewContainer).empty();
+        if (!results) {
+          return;
+        }
+        this.addImage(results);
+        this.buildPreviewHeader(suggestion);
+        this.buildResultsPreview(results);
+      });
+  }
+
+  private buildResultsPreview(results: IQueryResults): Promise<HTMLElement[]> {
+    const resultsContainer = this.buildResultsContainer();
+    this.previewContainer.appendChild(resultsContainer);
+    this.setupRenderer(resultsContainer);
+    this.buildResults(results).then(HTMLResult => {
+      if (!(HTMLResult.length > 0)) {
+        return;
+      }
+      this.updateResultPerRow(HTMLResult);
+      this.renderer.renderResults(HTMLResult, true, result => {});
+    });
+    return;
+  }
+
+  private updateResultPerRow(HTMLElements: HTMLElement[]) {
```

Comment:
why the upper case name for the variable ?

Also the name is relatively poor, im guessing it could be `rows: HTMLElement[]` ? Which gives more information about "what" the html elements represent.